GDJS/Runtime/pixi-renderers/pixi-object-effects-manager.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ * GDevelop JS Platform
+ * Copyright 2013-2021 Florian Rival (Florian.Rival@gmail.com). All rights reserved.
+ * This project is released under the MIT License.
+ */
+namespace gdjs {
+  import PIXI = GlobalPIXIModule.PIXI;
+
+  class PixiObjectEffectsManager {
+    protected _filters: Hashtable<Hashtable<PixiFiltersTools.Filter>>;
+
+    constructor() {
+      this._filters = new Hashtable();
+    }
+
+    update(runtimeObject: RuntimeObject, layer: Layer) {
+      const filters: Hashtable<
+        PixiFiltersTools.Filter
+      > | void = this._filters.get(runtimeObject.getName());
+      if (!filters) return;
+      const filterValues: PixiFiltersTools.Filter[] = [];
+      filters.values(filterValues);
+      for (const filter of filterValues) {
+        filter.update(filter.pixiFilter, layer);
+      }
+    }
+
+    /**
+     * Add a new effect on a runtime object, or replace the one
+     * with the same name.
+     * @param runtimeObject The runtime object
+     * @param effectData The data of the effect to add.
+     */
+    addEffect(
+      runtimeObject: RuntimeObject,
+      effectData: EffectData,
+      layer: Layer
```

Comment:
Don't pass the layer to addEffect or anywhere. Effects on objects should have no access to the layer at all.

Reply:
I was passing the layer because, the `update` function of the filters require `getElapsedTime()`, for example here:
https://github.com/4ian/GDevelop/blob/f3a49ad2cfd48efa6bed17565041094390a199b6/Extensions/Effects/glitch-pixi-filter.ts#L10-L18
The `getElapsedTime()` of `runtime Object` basically calls the function of  `layer`:
https://github.com/4ian/GDevelop/blob/f3a49ad2cfd48efa6bed17565041094390a199b6/GDJS/Runtime/runtimeobject.ts#L193-L197

Reply:
In the future, maybe some objects will have a different time scale than other, so it's better to use the function provided by the object :)

Consider the "layer" object passed to the effects as a new type of object, which should contain basically "getElapsedTime, getWidth, getHeight". Not sure how to call it in TypeScript and as the parameter, but it's probably something we can call "EffectTarget"/"effectTarget".

Reply:
Right, I think for the moment it can be a union type of gdjs.Layer and gdjs.RuntimeObject, but again, the `getElapsedTime()` 
 method is not equivalent at the moment for layers and runtimeobject (there's a difference of arguments). We might need some refactoring here.

Reply:
Ah right! Mmmm let's make this a TODO. We can think about changing what we pass to effects (rather than passing an entire layer or runtimeobject, we can pass a few useful functions, we'll see!)